packages/rulesets/src/arazzo/index.ts

```diff
@@ -0,0 +1,258 @@
+      given: '$',
+      then: {
+        function: arazzoStepRequestBodyValidation,
+      },
+    },
+    'arazzo-step-validation': {
+      description:
+        'Every step must have a valid "stepId" and an valid "operationId" or "operationPath" or "workflowId".',
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoStepValidation,
+      },
+    },
+    'arazzo-no-script-tags-in-markdown': {
+      description: 'Markdown descriptions must not have "<script>" tags.',
+      recommended: true,
+      given: '$..[description,title]',
+      then: {
+        function: pattern,
+        functionOptions: {
+          notMatch: '<script',
+        },
+      },
+    },
+    'arazzo-info-description': {
+      description: 'Info "description" should be present and non-empty string.',
+      recommended: true,
+      severity: 'warn',
+      given: '$',
+      then: {
+        field: 'info.description',
+        function: truthy,
+      },
+    },
+    'arazzo-info-summary': {
+      description: 'Info "summary" should be present and non-empty string.',
+      recommended: true,
+      severity: 'hint',
+      given: '$',
+      then: {
+        field: 'info.summary',
+        function: truthy,
+      },
+    },
+    'arazzo-source-descriptions-type': {
+      description: 'Source Description "type" should be present.',
+      recommended: true,
+      severity: 'warn',
+      given: '$.sourceDescriptions[*]',
+      then: {
+        field: 'type',
+        function: truthy,
+      },
+    },
+    'arazzo-workflow-workflowId': {
+      description: 'Workflow "workflowId" should follow the pattern "^[A-Za-z0-9_\\-]+$".',
+      recommended: true,
+      severity: 'warn',
+      given: '$.workflows[*]',
+      then: {
+        field: 'workflowId',
+        function: pattern,
+        functionOptions: {
+          match: '^[A-Za-z0-9_\\-]+$',
+        },
+      },
+    },
+    'arazzo-workflow-description': {
+      description: 'Workflow "description" should be present and non-empty string.',
+      recommended: true,
+      severity: 'warn',
+      given: '$.workflows[*]',
+      then: {
+        field: 'description',
+        function: truthy,
+      },
+    },
+    'arazzo-workflow-summary': {
+      description: 'Workflow "summary" should be present and non-empty string.',
+      recommended: true,
+      severity: 'hint',
+      given: '$.workflows[*]',
+      then: {
+        field: 'summary',
+        function: truthy,
+      },
+    },
+    'arazzo-step-description': {
+      description: 'Step "description" should be present and non-empty string.',
+      recommended: true,
+      severity: 'warn',
+      given: '$.workflows[*].steps[*]',
+      then: {
+        field: 'description',
+        function: truthy,
+      },
+    },
+    'arazzo-step-summary': {
+      description: 'Step "summary" should be present and non-empty string.',
+      recommended: true,
+      severity: 'hint',
```

Comment:
Description says should be present but it is a hint? Maybe change it to something like its recommended to have a summary?